lib/compat/experimental/blocks.php

```diff
@@ -160,3 +160,152 @@ function ( $response ) {
 	}
 }
 add_action( 'rest_api_init', 'gutenberg_rest_comment_set_children_as_embeddable' );
+
+if ( ! function_exists( 'gutenberg_get_comments_pagenum_link' ) ) {
+	/**
+	 * Refactor of core get_comments_pagenum_link function to make it work with comments blocks.
+	 *
+	 * It's used in the Comments Next Link and Comment Previous link blocks.
+	 *
+	 * @param WP_Block $block Block instance.
+	 * @param int  $pagenum Page number.
+	 *
+	 * @return string Returns the comments page number link URL.
+	 */
+	function gutenberg_get_comments_pagenum_link(  $block, $pagenum = 1 ) {
```

Comment:
Is the goal for this function to refactor and replace Core's `get_comments_pagenum_link()`?

If yes, it'll break backwards-compatibility due to the function parameter signature change of the first param being `$block` and missing the `$max_page` param.

If no, is the intend for this to be a new global function in Core that is only for block content? In that case, I'm thinking more about a Pagination API that supports both block and non-block content, thereby merging Core's and Gutenberg's functionality needs.

Reply:
Same question for the other functions in this PR.

Reply:
I've come to the same conclusion as @hellofromtonya.
IMO, if these new functions (i.e. `gutenberg_get_comments_pagenum_link`, `gutenberg_get_next_comments_link`, `gutenberg_get_previous_comments_link`) will have to be backported to Core later, we need to change their names to not confuse them with existing `get_comments_pagenum_link`, `get_previous_comments_link`, and `gutenberg_get_previous_comments_link` Core functions.

Reply:
I've come to the same conclusion as @hellofromtonya.
IMO, if these new functions (i.e. `gutenberg_get_comments_pagenum_link`, `gutenberg_get_next_comments_link`, `gutenberg_get_previous_comments_link`) will have to be backported to Core later, we need to change their names to not confuse them with existing `get_comments_pagenum_link`, `get_previous_comments_link`, and `gutenberg_get_previous_comments_link` Core functions.